### PR TITLE
[DO NOT MERGE]bugfix: Prevent Infinite Transaction Cookie Accumulation + Configuration Options

### LINKED ISSUE #1917: v4: Infinitely stacking cookies
### Checklist

- [x] The issue can be reproduced in the [nextjs-auth0 sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [x] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [x] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [x] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [x] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [x] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

- v4 sdk creates a new additional transactional cookie each time the unauthenticated user navigates to the app
- v4 logout does not remove said cookies

v3 does not create infinite cookies and does remove cookies on logout.

This eventually leads to a situation where the header of the request is too large.

### Reproduction

1. Be unauthenticated in your application (remove all cookies on that domain if you want)
2. Navigate to a route in your application (receive a cookie)
3. Navigate to another route in your application (receive another cookie)
4. Repeat as many times as you wish
5. Navigate to `/auth/logout` (receive another cookie, and cookies are not removed)

https://github.com/user-attachments/assets/21076f33-3c81-428b-b84b-4d566883cb4e

### Additional context

N/A

### nextjs-auth0 version

4.0

### Next.js version

15.1.6

### Node.js version

20.x

### LINKED ISSUE #2209: Transaction cookie not clean after usage lead to 413 or lambda size limit
### Checklist

- [x] The issue can be reproduced in the [nextjs-auth0 sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [x] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [x] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [x] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [x] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [x] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

I reopen an issue, because subject of https://github.com/auth0/nextjs-auth0/issues/1917 

I'm using version 4.6.0 of auth0/nextjs-auth0 and next 15.3.0 and I still have a side effect with __txn_ cookies are growing in certain circumstances.

Image

Logout work well to clean them 
But if you never logout (as a regular user does), you number of cookies grows and lead to reach limit of infrastructure.
I'm using a lambda deployment so the limit is 6Mb and the __txn_ cookie size is around 400 bytes... so after sometime (with additional cookies in the loop), user reach the limit and is block with "Request must be smaller than 6291456 bytes for the InvokeFunction operation" 

Cleaning the __txn_ solve the issue

We implemented successfully the workaround proposed in middleware to do the transaction cookie clean on any path except /auth https://github.com/auth0/nextjs-auth0/issues/1917#issuecomment-2677010176 

But it should be great to have a builtin fix in the library -> cleaning transaction cookie after use

### Reproduction

Not a clear simple scenario to reproduce
Navigate in application probably using multiple tabs and never logout

Can be reproduced adding manually __txn_ cookies directly in developer tools

### Additional context

_No response_

### nextjs-auth0 version

4.6.0

### Next.js version

15.3.0

### Node.js version

20.18.1

### PR DESCRIPTION
**Pull request by tusharpandey13**
Fixes transaction cookie accumulation that caused 413 errors when users repeatedly navigate to protected routes. Adds configurable cleanup with three new `AuthClient` options for different application needs.

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### 🔍 RCA
Each `startInteractiveLogin()` call created new `__txn_*` cookies but only deleted the matching callback cookie, leaving abandoned cookies from incomplete flows to accumulate until hitting HTTP header limits.

### 📋 Changes

Adds configurable transaction cookie management to prevent accumulation while supporting multi-tab scenarios. New `AuthClient` constructor options:

- `enableParallelTransactions: boolean` (default: `true`) - Controls single vs multi-transaction mode
- Added `maxAge` property to `TransactionCookieOptions` to control txn cookie maxage (3600s by default)

Usage example for single-transaction mode:
```typescript
new AuthClient({
  // ...required options
  transactionCookieOptions: {
    // ...options
    maxAge: 60 // 1 minute
  }
  enableParallelTransactions: false,
});
```

Changed `src/server/auth-client.ts`: Added configuration options and cleanup in `startInteractiveLogin()`
Added `tests/v4-infinitely-stacking-cookies.test.ts`: Comprehensive test coverage for all options

### 📎 References

Fixes: #1917
Fixes: #2209

### 🎯 Testing

**Automated:**
Added comprehensive test suite in `tests/v4-infinitely-stacking-cookies.test.ts` covering all configuration options, cleanup scenarios, and edge cases. Fixed CI test failure by providing explicit fetch mock and routes configuration instead of accessing private AuthClient properties.

**Manual:**
1. Navigate to protected route multiple times and verify ≤3 transaction cookies remain
2. Test multi-tab login scenarios work correctly with `enableParallelTransactions: true`
3. Test single-transaction mode with `enableParallelTransactions: false`
4. Run `pnpm test:unit` to verify all tests pass

### CODE REVIEW

File: src/server/auth-client.ts [hunk 1]
```diff
@@ -404,7 +444,24 @@ export class AuthClient {
 
     // Set response and save transaction
     const res = NextResponse.redirect(authorizationUrl.toString());
-    await this.transactionStore.save(res.cookies, transactionState);
+
+    // Apply transaction cookie management based on configuration
+    if (reqCookies) {
+      if (!this.enableParallelTransactions) {
+        // When parallel transactions are disabled, delete all existing transaction cookies
+        await this.transactionStore.deleteAll(reqCookies, res.cookies);
```
Comment (frederikprijck) [inline src/server/auth-client.ts]: Can we, instead of doing this, ensure the cookie name is not unique but we use a single transaction cookie name [here](https://github.com/auth0/nextjs-auth0/blob/d916c1823b69730c47971ed7a9026412a222a7be/src/server/transaction-store.ts#L75)?

File: src/server/transaction-store.ts [hunk 2]
```diff
@@ -36,11 +36,26 @@ export interface TransactionCookieOptions {
    * The path attribute of the transaction cookie. Will be set to '/' by default.
    */
   path?: string;
+  /**
+   * The expiration time for transaction cookies in seconds.
+   * If not provided, defaults to 1 hour (3600 seconds).
+   *
+   * @default 3600
+   */
+  maxAge?: number;
```
Comment (tusharpandey13) [inline src/server/transaction-store.ts:45]: make maxAge of txn cookie configurable

File: src/server/auth-client.ts (hunk 1 above)
Comment (tusharpandey13) [reply src/server/auth-client.ts]: done

**Comment by tusharpandey13** [comment]
This is irrelevant now, see https://github.com/auth0/nextjs-auth0/issues/1917#issuecomment-3117875981
